Sources/Tracing/Tracer.swift

```diff
@@ -152,55 +235,76 @@ extension Tracer {
     ///   - operationName: The name of the operation being traced. This may be a handler function, database call, ...
     ///   - baggage: The `Baggage` providing information on where to start the new ``Span``.
     ///   - kind: The ``SpanKind`` of the new ``Span``.
-    ///   - time: The time at which to start the new ``Span``.
+    ///   - clock: The clock to use as time source for the start time of the ``Span``
     ///   - function: The function name in which the span was started
     ///   - fileID: The `fileID` where the span was started.
     ///   - line: The file line where the span was started.
     ///   - operation: The operation that this span should be measuring
     /// - Returns: the value returned by `operation`
     /// - Throws: the error the `operation` has thrown (if any)
-    #if swift(>=5.7.0)
-    @_unsafeInheritExecutor
-    @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *)
+    @available(macOS 10.15, iOS 13, tvOS 13, watchOS 6, *) // for TaskLocal Baggage
+    public static func withSpan<T, Clock: TracerClock>(
+        _ operationName: String,
+        clock: Clock,
+        baggage: @autoclosure () -> Baggage = .current ?? .topLevel,
+        ofKind kind: SpanKind = .internal,
+        function: String = #function,
+        file fileID: String = #fileID,
+        line: UInt = #line,
+        _ operation: (any Span) async throws -> T
```

Comment:
Aren't we missing a `@_inheritActorContext` and `@Sendable` annotation here. Otherwise this could runs into warnings right?


```
actor Foo {
  var bar = 0 
  func foo() async {
    await withSpan {
      self.bar += 1
    }
  }
}
```

The downside of adding these annotations is that the story inside classes/structs is becoming worse because they don't have any actor context so the `@Sendable` constraint makes it impossible for them to access state even though the `operation` closure is run "inline".

In the end, we really need a productionized version of `@_unsafeInheritExecutor`. We are running into the same problem over in service-lifecycle.

Reply:
I checked and we don't get warnings for this 🤔